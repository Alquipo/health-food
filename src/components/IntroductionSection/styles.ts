import styled, { css } from 'styled-components'

import media from 'styled-media-query'

export const Wrapper = styled.section`
  ${({ theme }) => css`
    height: 100vh;
    form {
      margin-top: ${theme.spacings.xsmall};
    }

    ${media.lessThan('large')`
      height: auto;
    `};
  `}
`

export const Content = styled.div`
  display: flex;
  align-items: center;
  margin: 10rem auto;
  max-width: 120rem;

  ${media.lessThan('large')`
    margin-top: 0;
    padding: 10rem 0;
    justify-content: center;
  `};
`

export const FirstWrapper = styled.div`
  ${({ theme }) => css`
    flex: 1;

    button {
      margin-left: ${theme.spacings.xxsmall};
    }

    ${media.lessThan('large')`
      width: 100%;
      max-width: 55rem;
    `};
  `}
`

export const SecondWrapper = styled.div`
  flex: 2;

  ${media.lessThan('large')`
    display: none;
  `};
`

export const Title = styled.h1`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.huge};
    color: ${theme.colors.secondary};
    letter-spacing: 0.1rem;
    line-height: 1.3;

    ${media.lessThan('large')`
    font-size: ${theme.font.sizes.xxxlarge};

    `};
  `}
`

export const Image = styled.img`
  height: 100%;
  object-fit: cover;
  display: block;
  position: absolute;
  top: 0;
  right: 0;
  z-index: -1;

  ${media.lessThan('large')`
    display: none;
  `};
`
