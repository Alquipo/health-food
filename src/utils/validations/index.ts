import * as yup from 'yup'

const nascimentoRegex =
  /^((((19|[2-9]\d)\d{2})-(0[13578]|1[02])-(0[1-9]|[12]\d|3[01]))|(((19|[2-9]\d)\d{2})-(0[13456789]|1[012])-(0[1-9]|[12]\d|30))|(((19|[2-9]\d)\d{2})-02-(0[1-9]|1\d|2[0-8]))|(((1[6-9]|[2-9]\d)(0[48]|[2468][048]|[13579][26])|((16|[2468][048]|[3579][26])00))-02-29))$/g

const cpfRegex = /^\d{3}.?\d{3}.?\d{3}-?\d{2}$/

const cepRegex = /^([\d]{2})\.*([\d]{3})-*([\d]{3})/

const estadoRegex =
  /^(AC|AL|AP|AM|BA|CE|DF|ES|GO|MA|MT|MS|MG|PA|PB|PR|PE|PI|RJ|RN|RS|RO|RR|SC|SP|SE|TO)$/i

export const fieldsValidationSchema = yup.object({
  name: yup.string().required('Is not allowed to be empty'),
  birthDate: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(nascimentoRegex, 'must be a valid email'),
  cpf: yup
    .string()
    .max(14, 'Must be 14 characters')
    .required('Is not allowed to be empty')
    .matches(cpfRegex, 'Invalid CPF'),
  zipCode: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(cepRegex, 'Invalid ZipCode'),
  address: yup.string().required('Is not allowed to be empty'),
  district: yup.string().required('Is not allowed to be empty'),
  number: yup.number().required('Is not allowed to be empty'),
  city: yup.string().required('Is not allowed to be empty'),
  state: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(estadoRegex, 'Invalid State')
})
