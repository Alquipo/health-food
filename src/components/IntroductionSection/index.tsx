import * as S from './styles'

import { Formik, Form } from 'formik'

import { Search } from '@styled-icons/boxicons-regular'

import Button from '../Button'
import Menu from '../Menu'
import TextField from '../TextField'

type FormType = {
  text: string
}

const initialValues: FormType = {
  text: ''
}

export default function IntroductionSection() {
  async function handleSubmit() {
    return null
  }

  return (
    <>
      <S.Wrapper>
        <Menu />

        <S.Content>
          <S.FirstWrapper>
            <S.Title>Ready for Trying a new recipe?</S.Title>

            <Formik initialValues={initialValues} onSubmit={handleSubmit}>
              {() => (
                <Form style={{ display: 'flex' }}>
                  <TextField
                    placeholder="Search healthy recipes"
                    name="recipes"
                    type="text"
                    fullWidth
                  />
                  <Button type="submit" aria-label="Submit">
                    <Search size="24" />
                  </Button>
                </Form>
              )}
            </Formik>
          </S.FirstWrapper>
          <S.SecondWrapper></S.SecondWrapper>
        </S.Content>
      </S.Wrapper>
      <S.Image src="/img/Illustration.svg" alt="um peso e uma maçã" />
    </>
  )
}
