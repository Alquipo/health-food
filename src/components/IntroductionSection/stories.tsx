import { Story, Meta } from '@storybook/react'
import IntroductionSection from '.'

export default {
  title: 'IntroductionSection',
  component: IntroductionSection
} as Meta

export const Default: Story = () => <IntroductionSection />
