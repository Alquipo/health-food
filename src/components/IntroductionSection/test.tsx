import { fireEvent, render, screen, waitFor } from 'utils/test-utils'

import IntroductionSection from '.'

describe('<IntroductionSection/>', () => {
  it('should render a Introduction Section correctly', () => {
    const { container } = render(<IntroductionSection />)

    expect(
      screen.getByRole('heading', {
        name: /Ready for Trying a new recipe?/i
      })
    ).toBeInTheDocument()

    expect(screen.getByTestId(/recipes/i)).toBeInTheDocument()

    expect(screen.getByRole('button', { name: /submit/i })).toBeInTheDocument()

    expect(
      screen.getByRole('img', { name: /um peso e uma maçã/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should handle submit button ', () => {
    render(<IntroductionSection />)

    const handleSubmit = jest.fn()

    fireEvent.click(screen.getByLabelText(/submit/i))

    waitFor(() => {
      expect(handleSubmit).toHaveBeenCalled()
    })
  })
})
