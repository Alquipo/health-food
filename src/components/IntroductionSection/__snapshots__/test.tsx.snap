// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntroductionSection/> should render a Introduction Section correctly 1`] = `
.c0 {
  height: 100vh;
  padding: 0 1.5rem;
}

.c0 form {
  margin-top: 1.6rem;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 10rem auto;
  max-width: 120rem;
}

.c9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c9 button {
  margin-left: 0.8rem;
}

.c14 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c10 {
  font-size: 4.8rem;
  color: #1D164D;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  line-height: 1.3;
}

.c7 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c13 {
  padding: 1.6rem 2.4rem;
  background: #BADC58;
  color: #FFFFFF;
  border-radius: 0.6rem;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
  cursor: pointer;
  border: none;
  font-size: 1.8rem;
  font-weight: 600;
}

.c13:hover {
  -webkit-filter: brightness(0.95);
  filter: brightness(0.95);
}

.c1 {
  height: 10rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  margin: 0 auto;
  max-width: 120rem;
  padding: 0 1.5rem;
}

.c1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
}

.c2 {
  color: #BADC58;
  font-size: 3.2rem;
}

.c3 {
  color: #FFFFFF;
}

.c4 {
  margin-right: 2.4rem;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c4:hover {
  -webkit-filter: brightness(0.95);
  filter: brightness(0.95);
}

.c5 {
  border: none;
  text-transform: uppercase;
  font-size: 1.6rem;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
  padding: 0.8rem 1.6rem;
  background: #FFFFFF;
  border-radius: 0.6rem;
  color: #BADC58;
  cursor: pointer;
  font-weight: 600;
}

.c6 {
  background: transparent;
  border: none;
  cursor: pointer;
}

.c6 svg {
  fill: #1D164D;
}

.c11 {
  width: 100%;
  max-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  grid-area: recipes;
}

.c12 {
  display: block;
  width: 100%;
  padding: 1.6rem;
  border: solid 0.1rem #1D164D;
  border-radius: 0.6rem;
  font-size: 1.6rem;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
  font-size: 1.4rem;
}

.c12:focus,
.c12:hover {
  outline: none;
}

@media (max-width:1170px) {
  .c0 {
    height: auto;
  }
}

@media (max-width:1170px) {
  .c8 {
    margin-top: 0;
    padding: 10rem 0;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:1170px) {
  .c9 {
    width: 100%;
    max-width: 55rem;
  }
}

@media (max-width:1170px) {
  .c14 {
    display: none;
  }
}

@media (max-width:1170px) {
  .c10 {
    font-size: 3.6rem;
  }
}

@media (max-width:1170px) {

}

@media (max-width:768px) {
  .c3 {
    position: absolute;
    opacity: 0;
    pointer-events: none;
    top: 100%;
    right: 0;
    background: #BADC58;
    z-index: 20;
  }
}

@media (max-width:1170px) {
  .c4 {
    color: #9E9BAF;
  }
}

@media (max-width:768px) {
  .c4 {
    color: #FFFFFF;
    display: block;
    padding: 1.6rem 2.4rem;
    border-bottom: solid 0.1rem #FFFFFF;
    width: 100%;
  }
}

@media (max-width:1170px) {
  .c5 {
    background: #BADC58;
    color: #FFFFFF;
  }
}

@media (min-width:768px) {
  .c5 {
    padding: 1.6rem 2.4rem;
  }
}

@media (min-width:768px) {
  .c6 {
    display: none;
  }
}

<section
  class="c0"
>
  <menu
    class="c1"
  >
    <h1
      class="c2"
    >
      Healthy Food
    </h1>
    <nav
      aria-hidden="false"
      class="c3"
    >
      <a
        class="c4"
        href="#"
      >
        Healthy Recipes
      </a>
      <a
        class="c4"
        href="#"
      >
        Blog
      </a>
      <a
        class="c4"
        href="#"
      >
        Join
      </a>
      <a
        href="/sign-up"
      >
        <button
          class="c5"
        >
          Register
        </button>
      </a>
    </nav>
    <button
      class="c6"
    >
      <svg
        aria-hidden="true"
        aria-label="Open Menu"
        class="c7"
        fill="currentColor"
        focusable="false"
        height="40"
        viewBox="0 0 24 24"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 6h16v2H4zm0 5h16v2H4zm0 5h16v2H4z"
        />
      </svg>
    </button>
  </menu>
  <div
    class="c8"
  >
    <div
      class="c9"
    >
      <h1
        class="c10"
      >
        Ready for Trying a new recipe?
      </h1>
      <form
        action="#"
        style="display: flex;"
      >
        <div
          class="c11"
        >
          <input
            autocomplete="off"
            class="c12"
            data-testid="recipes"
            name="recipes"
            placeholder="Search healthy recipes"
            type="text"
            value=""
          />
        </div>
        <button
          aria-label="Submit"
          class="c13"
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="c7"
            fill="currentColor"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
            />
          </svg>
        </button>
      </form>
    </div>
    <div
      class="c14"
    />
  </div>
</section>
`;
